### PR TITLE
Include labels when loading attatchments, so UpdateLabels work

### PR DESCRIPTION
**Pull request by DownGoat**
can be removed

### CODE REVIEW

File: src/Equinor.ProCoSys.Completion.Infrastructure/Repositories/AttachmentRepository.cs [hunk 1]
```diff
@@ -1,20 +1,23 @@
...
     public Task<Attachment?> GetAttachmentWithFileNameForParentAsync(Guid parentGuid, string fileName, CancellationToken cancellationToken)
         => DefaultQuery.SingleOrDefaultAsync(
             a => a.ParentGuid == parentGuid && a.FileName == fileName,
             cancellationToken);
+
+    public async Task<Attachment> GetAttachmentWithLabelsAsync(Guid attachmentGuid, CancellationToken cancellationToken) 
+        => await DefaultQuery
+               .Include(a => a.Labels)
+               .SingleOrDefaultAsync(a => a.Guid == attachmentGuid, cancellationToken)
+           ?? throw new EntityNotFoundException<Attachment>(attachmentGuid.ToString());
```
Comment (janid1967) [inline src/Equinor.ProCoSys.Completion.Infrastructure/Repositories/AttachmentRepository.cs:22]: Please check if this improvement by using EntityNotFoundException<T> can be used elsewhere too

File: src/Equinor.ProCoSys.Completion.Infrastructure/Repositories/AttachmentRepository.cs [hunk 2]
```diff
@@ -1,20 +1,23 @@
...
-        : base(context, context.Attachments)
-    {
-    }
-
     public Task<Attachment?> GetAttachmentWithFileNameForParentAsync(Guid parentGuid, string fileName, CancellationToken cancellationToken)
         => DefaultQuery.SingleOrDefaultAsync(
             a => a.ParentGuid == parentGuid && a.FileName == fileName,
             cancellationToken);
+
+    public async Task<Attachment> GetAttachmentWithLabelsAsync(Guid attachmentGuid, CancellationToken cancellationToken)
```
Comment (janid1967) [inline src/Equinor.ProCoSys.Completion.Infrastructure/Repositories/AttachmentRepository.cs:18]: Add 3 unit tests:
* GetAttachmentWithLabelsAsync_WithLabels_ShouldReturnAttachmentWithLabels
* GetAttachmentWithoutLabelsAsync_WithLabels_ShouldReturnAttachmentWithEmptyLabelList
* GetAttachmentWithoutLabelsAsync_ForUnknownAttachment_ShouldThrowEntityNotFoundException

File: src/tests/Equinor.ProCoSys.Completion.Command.Tests/Attachments/AttachmentServiceTests.cs [hunk 3]
```diff
@@ -70,6 +70,8 @@ public void Setup()
 
         _attachmentRepositoryMock.GetAsync(_existingAttachment.Guid, default)
             .Returns(_existingAttachment);
+        _attachmentRepositoryMock.GetAttachmentWithLabelsAsync(_existingAttachment.Guid, default)
```
Comment (janid1967) [inline src/tests/Equinor.ProCoSys.Completion.Command.Tests/Attachments/AttachmentServiceTests.cs:73]: I guess mock above (GetAsync) can be removed now?

**Review by janid1967 — DISMISSED** [review]
See comments

File: src/Equinor.ProCoSys.Completion.Domain/EntityNotFoundException.cs [hunk 4]
```diff
@@ -11,3 +11,5 @@ public EntityNotFoundException(string message) : base(message)
     {
     }
 }
+
+public sealed class EntityNotFoundException<T>(string id) : Exception($"Could not find {typeof(T).Name} with Guid {id}");
```
Comment (janid1967) [inline src/Equinor.ProCoSys.Completion.Domain/EntityNotFoundException.cs:15]: Change type of input id from string to Guid

File: src/Equinor.ProCoSys.Completion.Infrastructure/Repositories/AttachmentRepository.cs (hunk 2 above)
Comment (DownGoat) [reply src/Equinor.ProCoSys.Completion.Infrastructure/Repositories/AttachmentRepository.cs:18]: Where do you want these tests?

File: src/Equinor.ProCoSys.Completion.Infrastructure/Repositories/AttachmentRepository.cs (hunk 1 above)
Comment (DownGoat) [reply src/Equinor.ProCoSys.Completion.Infrastructure/Repositories/AttachmentRepository.cs:22]: Added it to more places

File: src/Equinor.ProCoSys.Completion.Infrastructure/Repositories/AttachmentRepository.cs (hunk 2 above)
Comment (janid1967) [reply src/Equinor.ProCoSys.Completion.Infrastructure/Repositories/AttachmentRepository.cs:18]: In same testclass as existing tests for AttachmentRepository.cs exists
image

Comment (DownGoat) [reply src/Equinor.ProCoSys.Completion.Infrastructure/Repositories/AttachmentRepository.cs:18]: What am I supposed to test? That Microsoft has no bugs in LINQ?

**Review by siljehetl — APPROVED** [review]

**Pull request merged**